### ISSUE TITLE
XLS IR Visualizer falsely highlights edges that are not on the critical path

### ISSUE THREAD

**Issue by tcal-x** [google/xls#1065]
In some designs in the IR visualizer (https://google.github.io/xls/ir_visualization/), when critical path highlighting is selected, sometimes edges that are clearly not on the critical path get highlighted.  This seems to occur when there is a critical path
```
A -> B -> C -> D -> E -> F
```
but there also happens to be an edge for example 
```
B -> E
```
It seems that the rule is to highlight an edge whenever both its source and its destination are on the critical path.   This makes sense, but leads to non-critical edges such as `B -> E` getting highlighted.

This example is from cabac; the long edge on the right is the non-critical edge, in parallel with the true critical path that's to the left.

![cabac-crit](https://github.com/google/xls/assets/63030309/dbafbfed-2fc6-4a1b-8a16-44a85c740af6)

**Comment by tcal-x** [comment]
Extremely low priority!

**Comment by tcal-x** [comment]
This occurs even in the example screen cap shown on the IR Visualizer page: https://google.github.io/xls/ir_visualization/

There is a non-critical edge highlighted from the red 'sub' to the red 'shrl':

![crit-path](https://github.com/google/xls/assets/63030309/3aa76c8b-6213-40c3-8be6-f7deb0bd46f5)

The actual critical path goes through the bit_slice / sel / concat.
